pkg/process/checks/net_test.go

```diff
@@ -215,39 +223,76 @@ func TestNetworkConnectionBatchingWithDomains(t *testing.T) {
 		1: {
 			DnsStatsByQueryType: map[int32]*model.DNSStats{
 				int32(dns.TypeA): {
-					DnsTimeouts: 1,
+					DnsTimeouts: 4,
 				},
 			},
 		},
 		2: {
 			DnsStatsByQueryType: map[int32]*model.DNSStats{
 				int32(dns.TypeA): {
-					DnsTimeouts: 1,
+					DnsTimeouts: 5,
 				},
 			},
 		},
 	}
-	dns := map[string]*model.DNSEntry{}
+	dnsmap := map[string]*model.DNSEntry{}
 
 	cfg := config.NewDefaultAgentConfig(false)
 	cfg.MaxConnsPerMessage = 1
 
-	chunks := batchConnections(cfg, 0, conns, dns, "nid", nil, nil, domains, nil)
+	chunks := batchConnections(cfg, 0, conns, dnsmap, "nid", nil, nil, domains, nil)
 
 	assert.Len(t, chunks, 4)
 	total := 0
 	for i, c := range chunks {
 		connections := c.(*model.CollectorConnections)
 		total += len(connections.Connections)
+		domaindb := connections.GetDNSDatabase()
 		switch i {
 		case 0:
-			assert.Equal(t, []string{"", "", ""}, connections.Domains)
+			assert.Equal(t, len(domaindb), 0)
 		case 1:
-			assert.Equal(t, []string{"foo.com", "", ""}, connections.Domains)
+			assert.Equal(t, len(domaindb), 1)
+			assert.Equal(t, domains[0], domaindb[0])
+
+			// check for correctness of the data
+			conn := connections.Connections[0]
+			val, ok := conn.DnsStatsByDomainByQueryType[0]
+			assert.True(t, ok)
+			assert.Equal(t, val.DnsStatsByQueryType[int32(dns.TypeA)].DnsTimeouts, uint32(1))
 		case 2:
-			assert.Equal(t, []string{"foo.com", "", "baz.com"}, connections.Domains)
+			assert.Equal(t, len(domaindb), 2)
+			assert.Contains(t, domaindb, domains[0])
+			assert.Contains(t, domaindb, domains[2])
+			assert.NotContains(t, domaindb, domains[1])
+
+			indexzero := indexOf(domains[0], domaindb)
+			indextwo := indexOf(domains[2], domaindb)
+			conn := connections.Connections[0]
+			val, ok := conn.DnsStatsByDomainByQueryType[indexzero]
+			assert.True(t, ok)
+			assert.Equal(t, val.DnsStatsByQueryType[int32(dns.TypeA)].DnsTimeouts, uint32(2))
```

Comment:
Can we test with data that has multiple query types?

Reply:
Bump on this one.

Reply:
We can, but that's really beyond the scope of this PR.